x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/AllSupportedFieldsTestCase.java

```diff
@@ -518,10 +548,12 @@ private Matcher<String> expectedType(DataType type) {
             case BYTE, SHORT -> equalTo("integer");
             case HALF_FLOAT, SCALED_FLOAT, FLOAT -> equalTo("double");
             case NULL -> equalTo("keyword");
-            // Currently unsupported without TS command or KNN function
-            case AGGREGATE_METRIC_DOUBLE, DENSE_VECTOR ->
-                // TODO: Fix this once we know the node versions.
-                equalTo("unsupported");
+            case AGGREGATE_METRIC_DOUBLE, DENSE_VECTOR -> {
+                if (minVersion().supports(RESOLVE_FIELDS_RESPONSE_CREATED_TV) == false) {
```

Comment:
I think the correct TV to require here is the TV when the data type was officially released, that is, [this one](https://github.com/elastic/elasticsearch/blob/75e6e8a9d3b9a666daf303dd936e0fe2ddc8e2b9/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataType.java#L978-L980). Before that TV, the type was not officially released and thus, release builds shouldn't really support it.

It might not matter because there's no version branch that has `RESOLVE_FIELDS_RESPONSE_CREATED_TV` but doesn't have `ESQL_DENSE_VECTOR_CREATED_VERSION`, so our bwc tests won't pick this up. But it might still be better to use the "right" TV here so that future hackers will see which TV they need to plug into the test.

Reply:
I double checked and `RESOLVE_FIELDS_RESPONSE_CREATED_TV` comes *after* those. And we need both. So it's fine to check just `RESOLVE_FIELDS_RESPONSE_CREATED_TV` - but I'm adding a comment.

Reply:
I was wrong! The TVs I suggested are actually older. This is already correct.